google_guest_agent/run/run_test.go

```diff
@@ -0,0 +1,188 @@
+//  Copyright 2023 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package run
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+)
+
+// builds a set of data to be used in the tests
+func buildDataContent(t *testing.T) (string, error) {
+	t.Helper()
+	rootDir := path.Join(t.TempDir(), fmt.Sprintf("%d", time.Now().UnixNano()))
+	if err := Quiet("mkdir", "-p", rootDir); err != nil {
+		return rootDir, err
+	}
+
+	if err := os.WriteFile(path.Join(rootDir, "data"), []byte("random data"), 0644); err != nil {
+		return rootDir, err
+	}
+
+	return rootDir, nil
+}
+
+func TestQuietSuccess(t *testing.T) {
+	testDir, err := buildDataContent(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []string{
+		"grep -R data " + testDir,
+		"echo 'foobar' >> " + path.Join(testDir, "foobar"),
+		"rm -Rf " + path.Join(testDir, "foobar"),
+		"echo",
+	}
+
+	for _, curr := range tests {
+		t.Run(curr, func(t *testing.T) {
+			tokens := strings.Split(curr, " ")
+			if err := Quiet(tokens[0], tokens[1:]...); err != nil {
+				t.Errorf("failed command: %s, expected success.", curr)
+			}
+		})
+	}
+}
+
+func TestQuietFail(t *testing.T) {
+	testDir, err := buildDataContent(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []string{
+		"grep -R datax " + testDir,
+		"rm -R /root/data",
+	}
+
+	for _, curr := range tests {
+		t.Run(curr, func(t *testing.T) {
+			tokens := strings.Split(curr, " ")
+			if err := Quiet(tokens[0], tokens[1:]...); err == nil {
+				t.Errorf("command succeed, expected failure.")
```

Comment:
Can we update the failure messages to include input and identify the function?

Reply:
The input is identified in the `t.Run()` name, but it's harmful to repeat in the error msg. Will get it changed. Thanks.

Reply:
done. Thanks.